client/src/inside/components/page/EhrDialogForm.vue

```diff
@@ -34,8 +48,43 @@ export default {
       return this.tableDef.tableKey
     },
     groups () {
-      return this.tableDef.form ? this.tableDef.form.ehr_groups : []
+      let groups = this.tableDef.form ? this.tableDef.form.ehr_groups : []
+      if (this.acknowledgeSignature) {
+        return groups.filter(g => g.formCss !== 'record-header')
+      }
+      return groups
+    },
+    acknowledgeSignature () {
+      return (
+        this.isSigning &&
+        EhrDefs.getCaseStudyDataStatus(this.ehrHelp.pageKey) && 
+        this.hasPersonaData
+      )
+    }, 
+
+    hasCaseStudyData () {
+      return EhrDefs.getCaseStudyDataStatus(this.ehrHelp.pageKey) && 
+        this.hasPersonaData
+    },
+
+    disableSave () {
+      // disable save in case there is any case study data and the user hasn't acknowledged / confirmed
+      // it yet
+      return this.hasCaseStudyData ? !this.ackCaseStudyData : false
+    },
+    ackText () {
+      const persona = this.getPersonaData()
+      return `I hereby certify correct ${persona.persona}, ${persona.profession}. 
```

Comment:
This persona data may not exist, especially on assignments created before this feature. When I try the code these persona fields appear as undefined.  I think the hasCaseStudyData method is returning true when it should return false (on old assignments)

Reply:
I will do some more testing when it comes to legacy data. Thank you for pointing this out!